src/workerd/io/actor-cache-test.c++

```diff
@@ -950,7 +948,7 @@ KJ_TEST("ActorCache coalescing") {
         .withParams(CAPNP(keys = ["grault"]))
         .thenReturn(CAPNP(numDeleted = 0));
     mockTxn->expectCall("delete", ws)
-        .withParams(CAPNP(keys = ["garply", "fred", "waldo"]))
```

Comment:
Any idea what's with all these random reorderings of keys, both in this commit and in bb39bf72713bf13b7d781864de3a43ea014ade55?

Reply:
The "make entry removal easier" commit slightly changed `deleteAll()` https://github.com/cloudflare/workerd/pull/1814/commits/bb39bf72713bf13b7d781864de3a43ea014ade55#diff-52b552cc9d6477516b49e970aa23de236c7a1f7327b09c243c336445cbb2e18eR1989-R1993 so that now we just pick `Entry`s off the dirty list.

The reordering as seen from mock storage makes sense, because our dirtyList should look something like:
`foo, bar, grault, baz, garply` (where `foo` is least recent) prior to the `deleteAll` [here](https://github.com/cloudflare/workerd/pull/1814/commits/bb39bf72713bf13b7d781864de3a43ea014ade55#diff-80fa4859170178cc35404a2937001fc86b9430cada8aca5fe1cc3e3494509622R729).

We then switch it back in a later commit ([Make CountedDelete Simpler](https://github.com/cloudflare/workerd/pull/1814/commits/1669f0d0e8c8beceeac378adfbc6f1d164a7504b#diff-80fa4859170178cc35404a2937001fc86b9430cada8aca5fe1cc3e3494509622R753)) because rather than [giving an `Entry`](https://github.com/cloudflare/workerd/pull/1814/commits/1669f0d0e8c8beceeac378adfbc6f1d164a7504b#diff-52b552cc9d6477516b49e970aa23de236c7a1f7327b09c243c336445cbb2e18eL2129) a `CountedDelete` and adding it to the `CountedDelete`s entries [when we're flushing](https://github.com/cloudflare/workerd/pull/1814/commits/1669f0d0e8c8beceeac378adfbc6f1d164a7504b#diff-52b552cc9d6477516b49e970aa23de236c7a1f7327b09c243c336445cbb2e18eL2370), we just [immediately assign](https://github.com/cloudflare/workerd/pull/1814/commits/1669f0d0e8c8beceeac378adfbc6f1d164a7504b#diff-52b552cc9d6477516b49e970aa23de236c7a1f7327b09c243c336445cbb2e18eR2111-R2112) the `Entry` to the `CountedDelete`s entries in `putImpl()`. This essentially preserves the ordering as seen from the JS API, rather than whatever order things are in within the dirtyList.